### PR TITLE
Extract assessor to separated relation

### PR DESCRIPTION
**Pull request by jaleszek**
Decided to extract the Assessor as separated resource.
Arguments for:
- simplifying the structure - physical separation between admin and assessor views, arch. is more open for extensions and differences on behaviour between Asses/Admin, decreasing the amount of `if` in the views.
- following the REST principles a bit more
- no diverged logic for Admin/Assessor: assessor can be assigned to category, admin can't; assessor can be primary secondary or lead assessor in the same time; some logic seems to be unique for the assessor: way of retrieving the comments, form answers, assessment module among others, a lot logic seems to be unique for admins (notifications, dashboard, reports, list of assessors/admins)

against:
- duplicating some of the structures, especially at the beginning (BaseControllers, some views)
It can be overcome with using a bit of abstraction in the views, service objects in controllers, and assigning controllers to proper namespaces (looks like comments/form answer attachments can be moved to the global namespace as resources not strictly related with Admin namespace).
- some code smells: rendering the templates of admin resource in assessor resource (devise views)
- slightly longer development overhead as need to add functionality explicitly for Assessor.

### CODE REVIEW

File: app/models/assessor.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable, :confirmable
+
+  validates :first_name, :last_name, :role, presence: true
+  has_many :form_answer_attachments, as: :attachable
+
+  enumerize :role, in: %w[lead regular]
+
+  pg_search_scope :basic_search,
+                against: [
```
Comment (matthewford) [inline app/models/assessor.rb]: Align the parameters of a method call if they span more than one line.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      t.inet     :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Line is too long. [104/100]

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Extra blank line detected.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+class DeviseCreateAssessors < ActiveRecord::Migration
+  def change
+    create_table(:assessors) do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Unnecessary spacing detected.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Unnecessary spacing detected.<br>Put one space between the method name and the first argument.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Unnecessary spacing detected.<br>Put one space between the method name and the first argument.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.inet     :current_sign_in_ip
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Unnecessary spacing detected.<br>Put one space between the method name and the first argument.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      t.inet     :last_sign_in_ip
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Unnecessary spacing detected.<br>Put one space between the method name and the first argument.

File: db/migrate/20150227125140_devise_create_assessors.rb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+
+      t.timestamps
+    end
+
+    add_index :assessors, :email,                unique: true
```
Comment (matthewford) [inline db/migrate/20150227125140_devise_create_assessors.rb]: Unnecessary spacing detected.

File: db/migrate/20150227135432_add_confirmable_devise_fields_to_assessors.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+class AddConfirmableDeviseFieldsToAssessors < ActiveRecord::Migration
+  def change
+    change_table(:assessors) do |t|
+      t.string   :confirmation_token
```
Comment (matthewford) [inline db/migrate/20150227135432_add_confirmable_devise_fields_to_assessors.rb]: Unnecessary spacing detected.<br>Put one space between the method name and the first argument.

File: spec/models/assessor_spec.rb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Assessor, :type => :model do
```
Comment (matthewford) [inline spec/models/assessor_spec.rb]: Use the new Ruby 1.9 hash syntax.

File: spec/models/assessor_spec.rb [hunk 12]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (matthewford) [inline spec/models/assessor_spec.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

**Comment by jaleszek** [comment]
Remarkable new fact #1 

`Basically there can be any permutation
i.e. lead on one category, regular on another
a lead can be a primary or secondary in the same category that they lead`.

So actually I will have to remove the `role` from the Assessor model as well.

File: db/migrate/20150227125140_devise_create_assessors.rb (hunk 9 above)
Comment (jaleszek) [inline db/migrate/20150227125140_devise_create_assessors.rb]: please ignore

File: db/migrate/20150227125140_devise_create_assessors.rb (hunk 8 above)
Comment (jaleszek) [inline db/migrate/20150227125140_devise_create_assessors.rb]: ignore as well

**Comment by RuslanHamidullin** [comment]
Looks fine for me :+1:

**Comment by jaleszek** [comment]
@RuslanHamidullin thanks : - )

**Comment by RuslanHamidullin** [comment]
:beers:

**Pull request merged**